controllers/urls.js

```diff
@@ -1,16 +1,49 @@
-const express = require('express')
-const Url = require('../models/Url')
+const Url = require('../models/Url');
+const generateRandomString = require('../middleware/randomString');
+const { BASEURL, HTTP_ERRORS } = require('../constant');
 
-const createShortUrl = async(req, res) => {
-    const url = await Url.create({...req.body})
-    res.json(url)
-}
+const createShortUrl = async (req, res) => {
+  const { customName, originalUrl } = req.body;
+  let shortUrls;
 
-const customShortUrl = async(req, res)  => {
-    res.send('custom name for short url')
-}
+  try {
+    if (customName) {
+      const customNameWithDashes = customName.split(' ').join('-');
+      shortUrls = `${BASEURL}/${customNameWithDashes}`;
+      const existingCustomName = await Url.findOne({ customName: customNameWithDashes });
+      if (existingCustomName) {
+        return res.status(HTTP_ERRORS.BAD_REQUEST).json({ errors: 'Custom name already exists' });
+      }
+    } else {
+      const randomString = generateRandomString(5);
+      shortUrls = `${BASEURL}/${randomString}`;
+    }
+
+    const existingRandomString = await Url.findOne({ shortUrl: shortUrls });
+    if (existingRandomString) {
+      return res.status(HTTP_ERRORS.BAD_REQUEST).json({ error: 'This string already exists' });
+    }
+    const existingLongUrl = await Url.findOne({ originalUrl });
+    if (existingLongUrl) {
+      return res.status(HTTP_ERRORS.OK).json({ shortUrl: shortUrls, });
+    }
+    const url = await Url.create({
+      ...req.body,
+      shortUrl: shortUrls,
+      customName
+    });
+    res.status(HTTP_ERRORS.OK).json({ message: 'Url shortened succesfully', data: { shortUrl: shortUrls } });
```

Comment:
201 HTTP code is more suited to post routes that create data. Please add this to the HTTP STATUS constant

```suggestion
    res.status(HTTP_ERRORS.CREATED).json({ message: 'Url shortened succesfully', data: { shortUrl: shortUrls } });
```

Reply:
done on e8f6453